test/specs/data_sources/model/TraitDataVariable.ts

```diff
@@ -0,0 +1,254 @@
+            },
+          },
+        ],
+      })[0];
+
+      const input = cmp.getEl();
+      expect(input?.getAttribute('placeholder')).toBe('test-value');
+      expect(cmp?.getAttributes().placeholder).toBe('test-value');
+
+      const testDs = dsm.get(inputDataSource.id);
+      testDs.getRecord('id1')?.set({ value: 'new-value' });
+
+      expect(input?.getAttribute('placeholder')).toBe('new-value');
+      expect(cmp?.getAttributes().placeholder).toBe('new-value');
+    });
+
+    test('component updates with data-variable value', () => {
+      const inputDataSource: DataSourceProps = {
+        id: 'test-input',
+        records: [{ id: 'id1', value: 'test-value' }],
+      };
+      dsm.add(inputDataSource);
+
+      const cmp = cmpRoot.append({
+        tagName: 'input',
+        traits: [
+          'name',
+          'type',
+          {
+            type: 'text',
+            label: 'Value',
+            name: 'value',
+            value: {
+              type: DataVariableType,
+              value: 'default',
+              path: `${inputDataSource.id}.id1.value`,
+            },
+          },
+        ],
+      })[0];
+
+      const input = cmp.getEl();
+      expect(input?.getAttribute('value')).toBe('test-value');
+      expect(cmp?.getAttributes().value).toBe('test-value');
+
+      const testDs = dsm.get(inputDataSource.id);
+      testDs.getRecord('id1')?.set({ value: 'new-value' });
+
+      expect(input?.getAttribute('value')).toBe('new-value');
+      expect(cmp?.getAttributes().value).toBe('new-value');
+    });
+  });
+
+  describe('checkbox input component', () => {
+    test('component initializes and updates data-variable value', () => {
+      const inputDataSource: DataSourceProps = {
+        id: 'test-checkbox-datasource',
+        records: [{ id: 'id1', value: 'true' }],
+      };
+      dsm.add(inputDataSource);
+
+      const cmp = cmpRoot.append({
+        type: 'checkbox',
+        tagName: 'input',
+        attributes: { type: 'checkbox', name: 'my-checkbox' },
+        traits: [
+          {
+            type: 'checkbox',
+            label: 'Checked',
+            name: 'checked',
+            value: {
+              type: 'data-variable',
+              value: 'false',
+              path: `${inputDataSource.id}.id1.value`,
+            },
+            valueTrue: 'true',
+            valueFalse: 'false',
+          },
+        ],
+      })[0];
+
+      const input = cmp.getEl() as HTMLInputElement;
+      expect(input?.checked).toBe(true);
+      expect(input?.getAttribute('checked')).toBe('true');
+
+      const testDs = dsm.get(inputDataSource.id);
+      testDs.getRecord('id1')?.set({ value: 'false' });
+
+      expect(input?.getAttribute('checked')).toBe('false');
+      // Not syncing - related to
+      // https://github.com/GrapesJS/grapesjs/discussions/5868
+      // https://github.com/GrapesJS/grapesjs/discussions/4415
+      // https://github.com/GrapesJS/grapesjs/pull/6095
```

Comment:
Related to @artf comment in this PR: 

* https://github.com/GrapesJS/grapesjs/pull/6018/files#r1730402702